guides/ssl-certificates.md

```diff
@@ -0,0 +1,216 @@
+following:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloudflare-api-key-secret
+  namespace: coder # Your Coder deployment namespace
+type: Opaque
+stringData:
+    api-key: "" # Your Cloudflare API token (from earlier)
+
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Issuer
+metadata:
+  name: letsencrypt
+  namespace: coder # Your Coder deployment namespace
+spec:
+  acme:
+    email: "" # Your email address (given to Let's Encrypt)
+    server: "https://acme-v02.api.letsencrypt.org/directory"
+    privateKeySecretRef:
+      name: letsencrypt-account-key
+    solvers:
+    - dns01:
+        cloudflare:
+          email: "" # Your Cloudflare email address
+          apiTokenSecretRef:
+            name: cloudflare-api-key-secret
+            key: api-key
+
+    # This section denotes which domains to use this issuer for. If you didn't
+    # limit which zones the API token had access to, you may wish to remove
+    # this section.
+    selector:
+      dnsZones:
+         # Only use this issuer for the domain example.com and its subdomains.
+         - 'example.com'
+```
+
+### ClusterIssuers
+
+cert-manager has a concept of **Issuer** (which are per-namespace) or
+**ClusterIssuer** (which are global to the entire cluster). If you plan on using
+cert-manager only for Coder, you may choose to use the **Issuer** configuration
+above. If you want to use a **ClusterIssuer** instead, you'll need to make the
+following changes:
+
+Change the namespace of the secret to **cert-manager** Change the kind of the
+**Issuer** to **ClusterIssuer** Remove the namespace of the **ClusterIssuer**
+
+For further information, see [Setting Up
+Issuers](https://docs.cert-manager.io/en/release-0.8/tasks/issuers/index.html).
+
+Read the comments and fill out the blanks. Once you're done, you can go ahead
+and apply that to your cluster using:
+
+```bash
+$ kubectl apply -f issuer.yaml
+
+secret/cloudflare-api-key-secret created
+issuer.cert-manager.io/letsencrypt created
+```
+
+## Step 3: Issue Certificates for Coder
+
+In a text editor, create a new file called **certificates.yaml** and paste the
+following (make sure you modify where necessary based on the comments):
+
+```yaml
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: coder-root
+  namespace: coder # Your Coder deployment namespace
+spec:
+  secretName: coder-root-cert
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  dnsNames:
+    - "coder.example.com" # Your base domain for Coder
+  issuerRef:
+    name: letsencrypt
+    kind: Issuer
+
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: coder-devurls
+  namespace: coder # Your Coder deployment namespace
+spec:
+  secretName: coder-devurls-cert
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  dnsNames:
+     - "*.coder.example.com" # Your dev URLs wildcard subdomain
+issuerRef:
+  name: letsencrypt
+  kind: Issuer
+```
```

Comment:
This step isn't necessary and the certificates can be created automatically via the ingress resource. (ref: https://cert-manager.io/docs/usage/ingress/). 

To do this in Coder, in the step 4 yaml, use:

```yaml
ingress:
  additionalAnnotations:
  - 'cert-manager.io/cluster-issuer: nameofclusterissuer'
```

Following this example, the name of the issuer/cluster issuer is "letsencrypt." I have only tested this with a Cluster Issuer, but according to the cert-manager docs `cert-manager.io/issuer` can be used for an Issuer resource, which this tutorial uses instead.

cert-manager's ingress guide (linked above) as well as this separate doc have been SO helpful for me troubleshooting this stuff: https://cert-manager.io/docs/faq/troubleshooting/

cc @mterhar to include anything I may have missed.

Reply:
@bpmct I think I got this section correct, but could you please take a second look? Thanks!

Reply:
@khorne3 So your tutorial is correct and works, but I am pretty sure it isn't the recommended way to add Certificates.

Instead, they can be automatically created via the ingress resource when those three lines are added to the helm chart file.